### ISSUE TITLE
change(gridsome): add redirects from deprecated singer-specific detail pages

### ISSUE THREAD

**Issue by aaronsteers** [meltano/hub#795]
We previously hosted Singer-specific detail pages for extractors and loaders.

We can redirect all of these endpoints to the generic plugin endpoints.

E.g.

- `https://hub.meltano.com/taps/postgres` -> `https://hub.meltano.com/extractors/postgres` 
  - or `https://hub.meltano.com/extractors/tap-postgres`
- `https://hub.meltano.com/taps/postgres--singer-io` -> `https://hub.meltano.com/extractors/tap-postgres--singer-io`

**Comment by rwfeather** [comment]
Pasting from another comment I made:

> @rwfeather - Is there a simple means to add these redirects?

- Update the netlify config `netlify.toml` with a [redirect rule](https://docs.netlify.com/routing/redirects/#syntax-for-the-netlify-configuration-file) using a [splat in the path](https://docs.netlify.com/routing/redirects/redirect-options/#splats) 
- Create "pages" that have the desired path (`/singer/taps/*`, `/singer/target/*`) while using the same underlying component to render the page. You can see an example of this method in `gridsome.server.js 187:219`

```js
  // Create default variant pages
  api.createPages(async ({ createPage, graphql }) => {
    const defaultPlugins = await graphql(`...`)

    Object.keys(defaultPlugins.data).forEach((query) => {
      // For each default plugin we create an extra page one-level up
      defaultPlugins.data[query].edges.forEach(({ node }) => {
        createPage({
          // Remove the variant part of the path
          path: node.path.split("--")[0],
          // And send it to the same rendering template
          component: `./src/templates/Plugins.vue`,
          context: {
            id: node.id,
            // This is the path given to the `page-query` to do the lookup
            path: node.path,
            name: node.name,
          },
        });
      });
    });
```

**Comment by aaronsteers** [comment]
As noted in https://github.com/meltano/hub/issues/793#issuecomment-1254240358:

> @rwfeather - We are approved to remove these:
> 
> 1. The JSON API endpoints for taps and targets. (Users will be recommended to use the newer API which covers _all_ meltano plugin types.)
> 2. The Singer tap/target detail pages. (Redirect to 'extractor'/'loader' equivalents.)
>    
>    * Related follow-on: [feat(gridsome): add "standalone usage" section to main plugins pages #821](https://github.com/meltano/hub/issues/821)

**Comment by rwfeather** [comment]
Resolved in #824
